cmd/dataset_tokenizer/dataset_tokenizer.go

```diff
@@ -492,7 +492,12 @@ func WriteContexts(outPath string, nextContext ContextsIterator,
 				close(contexts)
 				break
 			} else {
-				contexts <- *context
+				if sampling == 100 {
```

Comment:
Some thoughts here.
* This could be combined with the following statement, into `sampling == 100 || rand.Intn(100) < sampling`
* Sampling is intended to get a % of the samples. Given this, what would the minimum sample size be for this to produce an approximation of the desired sampling percentage?
* Does sampling need to be random?

Reply:
accurate within 5% for large files now